File: Ejercicios en clase/Colas/ColaConPrioridadAcotada.java
import java.util.Arrays;

public class ColaConPrioridadAcotada<E> {
    private ColaADT<E>[] colas;
    private int maxPrioridad;

    public ColaConPrioridadAcotada(int maxPrioridad) {
        this.maxPrioridad = maxPrioridad;
        this.colas = new ColaADT[maxPrioridad];
        for (int i = 0; i < this.maxPrioridad + 1; i++) {
            this.colas[i] = new ColaADT<>();
        }
    }

    public int longitud() {
        int total = 0;
        for (int i = 0; i < maxPrioridad + 1; i++) {
            total += this.colas[i].longitud();
        }
        return total;
    }

    public boolean estaVacia() {
        return this.longitud() == 0;
    }

    public void encolar(int prioridad, E elemento) {
        if (prioridad >= 1 && prioridad <= this.maxPrioridad + 1) {
            this.colas[prioridad].encolar(elemento);
        }
    }

    public void desEncolar(int prioridad, E elemento) {
        if (this.estaVacia()) {
            System.out.println("No hay mas elementos");
        } else {
            /*
             * while (condition) {
             * 
             * }
             */
        }
    }

    @Override
    public String toString() {
        return "ColaConPrioridadAcotada [colas=" + Arrays.toString(colas) + ", maxPrioridad=" + maxPrioridad + "]";
    }

}

File: Tarea-8/ColaConPrioridadAcotada.py
from ColasADT import Queue


class ColaConPrioridadAcotada:
    def __init__(self, max_prioridad):
        self.max_prioridad = max_prioridad
        self.colas = [Queue() for _ in range(max_prioridad + 1)]

    def longitud(self):
        total = 0
        for cola in self.colas:
            total += cola.longitud()
        return total

    def esta_vacia(self):
        return self.longitud() == 0

    def encolar(self, prioridad, elemento):
        if 1 <= prioridad <= self.max_prioridad:
            self.colas[prioridad].encolar(elemento)

    def des_encolar(self):
        if self.esta_vacia():
            print("No hay más elementos")
        else:
            for cola in self.colas:
                if not cola.est_vacia():
                    return cola.des_encolar()
        return None

    def __str__(self):
        return f"ColaConPrioridadAcotada(colas={self.colas}, max_prioridad={self.max_prioridad})"
